src/extension/sdk/utils.ts

```diff
@@ -286,7 +286,7 @@ export class SdkUtils {
 		// If we're a Flutter workspace but we couldn't get the version, try running Flutter to initialise it first.
 		// Do this before searching for the Dart SDK, as it might download the Dart SDK we'd like to find.
 		if (hasAnyFlutterProject && flutterSdkPath && !workspaceConfig.skipFlutterInitialization) {
-			const flutterNeedsInitializing = !getSdkVersion(this.logger, { sdkRoot: flutterSdkPath, versionFile: workspaceConfig?.flutterVersionFile })
+			const flutterNeedsInitializing = !getSdkVersion(this.logger, { sdkRoot: flutterSdkPath, versionFile: workspaceConfig?.flutterVersionFile, isLatest: workspaceConfig?.isFlutterVersionLatest })
 				|| !fs.existsSync(path.join(flutterSdkPath, "bin/cache/dart-sdk"));
```

Comment:
It may be better to skip the call to `getSdkVersion` here and just use the flag directly - or even return the version directly in the workspace config instead of a flag:

```ts
const flutterVersion = workspaceConfig?.flutterVersion ?? getSdkVersion(...);
const flutterNeedsInitializing = !flutterVersion || !fs.existsSync(path.join(flutterSdkPath, "bin/cache/dart-sdk");
```